botocore/endpoint.py

```diff
@@ -234,41 +248,39 @@ def create_endpoint(self, service_model, region_name=None, is_secure=True,
                         response_parser_factory=None, timeout=DEFAULT_TIMEOUT):
```

Comment:
Yeah, I'm in favor of refactoring EndpointCreator a bit by removing endpoint and other unnecessary arguments. I take it you are too, so I'll make these changes.